src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Helpers/RemoveUnnecessaryImports/AbstractUnnecessaryImportsProvider.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.RemoveUnnecessaryImports
+{
+    internal abstract class AbstractUnnecessaryImportsProvider<T>
```

Comment:
Core `GetUnnecessaryImports` functionality is separated out from the `RemoveUnnecessaryImportsService`, which had functionality to both compute imports and remove unnecessary imports. No semantic code changes.